code/game/objects/items/weapons/sword_of_truth.dm

```diff
@@ -63,13 +64,25 @@
 				var/obj/item/weapon/implant/core_implant/I = M.get_core_implant(/obj/item/weapon/implant/core_implant/cruciform)
 				if(I && I.active && I.wearer)
 					continue
+				M.stats.addTempStat(STAT_TGH, -STAT_LEVEL_ADEPT, 45 SECONDS)
+				M.stats.addTempStat(STAT_VIG, -STAT_LEVEL_ADEPT, 45 SECONDS)
+				M.stats.addTempStat(STAT_ROB, -STAT_LEVEL_ADEPT, 45 SECONDS)
+				M.stats.addTempStat(STAT_BIO, -STAT_LEVEL_ADEPT, 45 SECONDS)
+				M.stats.addTempStat(STAT_COG, -STAT_LEVEL_ADEPT, 45 SECONDS)
+				M.stats.addTempStat(STAT_MEC, -STAT_LEVEL_ADEPT, 45 SECONDS)
```

Comment:
two things, 1 the flashbag proc already reduces the stats of the affected person.
2. specifies the source of the temporary stats, this is done by the fourth argument of the addTempStat proc.